tests/rules/method.js

```diff
@@ -456,5 +465,25 @@ eslintTester.run("method", rule, {
                 }
             ]
         },
+        { // admittedly, this doesnt make a lot of sense, since the func doesnt return a promise
+            code: "async () => await foo.insertAdjacentHTML('afterend', blah);",
+            parserOptions: { ecmaVersion: 2020 },
+            errors: [
+                {
+                    message: "Unsafe call to foo.insertAdjacentHTML for argument 1",
+                    type: "CallExpression"
+                }
+            ]
+        },
+        { // admittedly, this doesnt make a lot of sense, since the func doesnt return a promise
+            code: "async () => (await foo.insertAdjacentHTML('afterend', blah))();",
+            parserOptions: { ecmaVersion: 2020 },
+            errors: [
+                {
+                    message: "Unsafe call to foo.insertAdjacentHTML for argument 1",
+                    type: "CallExpression"
+                }
+            ]
+        }
```

Comment:
Do we want to also add something like:
```
        ...,
        {
            code: "async () => (await foo)().insertAdjacentHTML('afterend', blah);",
            parserOptions: { ecmaVersion: 2020 },
            errors: [
                {
                    message: "Unsafe call to (await foo)().insertAdjacentHTML for argument 1",
                    type: "CallExpression"
                }
            ]
        }
        ...
```

 explicitly in the list of the one we expect to be reported as linting error? 

The current implementation seems to be already handling it as expected, and so the main reason for adding it would be to just make sure we keep to report it as such in the future (in case of further changes or refactoring).